samples/hssi/hssi_200g_400g_cmd.h

```diff
@@ -0,0 +1,310 @@
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+#include "hssi_cmd.h"
+#include <cmath>
+#include <unistd.h>
+
+// The intention of this utility is to demonstrate simple use of the 200G/400G Ethernet subsystem so that a 
+// user may add their own complexity on top.
+
+// Code structure:
+// This class inherits hssi_cmd (which in turn inherits 'command') as part of the afu-test framework.
+// It implements a specific 'command' (which is essentially like a test / sequence of operations) to 
+// exercise the 200G-400G F-Tile HSSI Sub-System.
+
+// HSSI = "High Speed Serial Interface" which is better referred to as "Ethernet".
+// It expects a specific AFU (accelerator function unit) to be present in the FPGA (specified by afu_id()).
+// This corresponding AFU instantiates a traffic generator (TG) to transmit/receive
+// packet data to the HSSI-SS.
+
+// The run() function talks to the TG to create some test traffic flow and measure throughput.
+
+// The TG has very limited control -- you can only control the number of packets sent.
+// The following terms are synonymous: traffic generator, traffic controller, packet client.
+// The TG has its own control/status register (CSR) map that is abstracted behind a "Mailbox".
+// The Mailbox presents a small set of command registers that accept address, data, control.
+// Through these, the user can read/write the abstracted register space.
+// The Mailbox CSR address space is defined with the hssi_afu() class.
+// The Mailbox may have multiple ports (also called 'channels') where each port provides a window
+// into a distinct abstracted address space. In other words, the AFU may instantiate multiple
+// TGs where each TG is accessed through a separate Mailbox port.
+
+// It is assumed that a 200G AFU uses two Mailbox channels (i.e. two traffic generators) and a 
+// 400G AFU uses one channel.
+
+// Registers in the AFU CSR space. Other registers, in this space, are defined in hssi_afu.h
+#define CSR_AFU_400G_TG_EN                0x0058
+
+// 200G/400G traffic generator registers. This address space is accessed indirectly through the Mailbox
+// The CSR map is defined in a spreadsheet, located here at the time of this writing: 
+// https://github.com/OFS/ofs-fim-common/tree/release/ofs-2023.2/src/common/he_hssi/HE_HSSI_TC_200_400G.xls
```

Comment:
I think it's very helpful to link to the spreadsheet since it's in a different repo and will be otherwise hard to find.
But I see how this link refers specifically to the 2023.2 release, which may be too specific.